crates/core/src/db/queries/collections.rs

```diff
@@ -0,0 +1,70 @@
+//! Query utilities for collections.
+
+use anyhow::Context;
+use diesel::{
+    pg::Pg,
+    sql_types::{Array, Integer, Nullable, Text},
+    types::ToSql,
+    RunQueryDsl,
+};
+
+use crate::{
+    db::{custom_types::OrderDirection, models::Nft, Connection},
+    error::Result,
+};
+
+/// Query collections ordered by volume
+///
+/// # Errors
+/// returns an error when the underlying queries throw an error
+pub fn by_volume(
+    conn: &Connection,
+    addresses: impl ToSql<Nullable<Array<Text>>, Pg>,
+    order_direction: OrderDirection,
+    limit: impl ToSql<Integer, Pg>,
+    offset: impl ToSql<Integer, Pg>,
+) -> Result<Vec<Nft>> {
+    diesel::sql_query(make_by_volume_query_string(order_direction))
+        .bind(addresses)
+        .bind(limit)
+        .bind(offset)
+        .load(conn)
+        .context("Failed to load collections by volume")
+}
+
+fn make_by_volume_query_string(order_direction: OrderDirection) -> String {
+    format!(
+        r"
+        WITH volume_table (collection, volume) as (
+            SELECT metadata_collection_keys.collection_address AS collection, SUM(purchase_receipts.price) AS volume
```

Comment:
purchases table instead of purchase_receipts to be used here so that we can get collections for marketplaces which doesn't use receipts